test/src/main/java/io/strimzi/test/k8s/Exec.java

```diff
@@ -9,111 +9,285 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
-import java.util.ArrayList;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.CompletableFuture;
```

Comment:
The rest of the project is using vertx `Futures`. It's not clear to me whether whatever you're doing with the `CompletableFuture` API here cannot be done using vertx, but if possible we should avoid using two different async APIs in the same project.